src/provider/model/AppDirectory.ts

```diff
@@ -1,25 +1,45 @@
-import {injectable} from 'inversify';
+import {injectable, inject} from 'inversify';
+import {MaskWatch} from 'openfin-service-config/Watch';
+import {Scope} from 'openfin-service-config/Types';
+import {ScopedConfig} from 'openfin-service-config';
 
+import {Inject} from '../common/Injectables';
 import {Application, AppName} from '../../client/directory';
 import {AppIntent} from '../../client/main';
+import {ConfigurationObject} from '../../../gen/provider/config/fdc3-config';
+
+import {ConfigStore} from './ConfigStore';
 
 enum StorageKeys {
     URL = 'fdc3@url',
     APPLICATIONS = 'fdc3@applications'
 }
 
 // Demo development app directory url
-const devUrl = 'http://localhost:3923/provider/sample-app-directory.json';
-
+export const devAppDirectoryUrl = 'http://localhost:3923/provider/sample-app-directory.json';
 
 @injectable()
 export class AppDirectory {
     private _directory: Application[] = [];
     private _url!: string;
 
-    public constructor() {
-        // Set default values
-        this._url = devUrl;
+    public constructor(@inject(Inject.CONFIG_STORE) configStore: ConfigStore) {
+        if (process.env.NODE_ENV === 'development') {
+            // Set the application directory to our local copy during development.  In production it will be an empty string.
+            configStore.config.add({level: 'desktop'}, {applicationDirectory: devAppDirectoryUrl});
+        }
+
+        this._url = configStore.config.query({level: 'desktop'}).applicationDirectory;
+
+        // Adding the follow watch logic as there is a race condition with the loading of the application directory.
+        // This logic will set the URL in the store as soon as it is retrieved.
```

Comment:
What's the race condition? Could we make sure configStore `initialize` doesn't resolve until these it's ready to have the `applicationDirectory` queried

Reply:
So the comment here was a bit wrong.  Its not really a race condition, but rather a mix of a feature and a need in testing.  Basically, if the desktop owner config changes after the initial load, this will pick up any change to the `applicationDirectory`.  This is more or a less of a feature to come.  TBD.  Following similar in testing, we manually set the `applicationDirectory` after the store is initialized so this is also needed to pick up that change. 

I have moved this to `AppDirectory.init` method and added an await for the `ConfigStore` there as it still makes sense to do that.